crates/sui/src/sui_commands.rs

```diff
@@ -1239,3 +1241,26 @@ pub fn parse_host_port(
         format!("{default_host}:{default_port_if_missing}").parse::<SocketAddr>()
     }
 }
+
+/// Warn the user if the CLI falls behind more than 2 protocol versions.
+async fn check_protocol_version_and_warn(context: &mut WalletContext) -> Result<(), anyhow::Error> {
+    let client = context.get_client().await?;
+    let protocol_cfg = client.read_api().get_protocol_config(None).await?;
+    let on_chain_protocol_version = protocol_cfg.protocol_version.as_u64();
+    let cli_protocol_version = ProtocolConfig::get_for_max_version_UNSAFE()
+        .version
+        .as_u64();
```

Comment:
I *think* this could just be `ProtocolVersion::MAX`. Otherwise LGTM

Reply:
Unfortunately, from what I can tell we can only call `max()` on `ProtocolConfig` if we have the actual object/reference to it, which we don't. If you know a way how to avoid the `get_for_max_version_UNSAFE()`, happy to learn!

Reply:
> I _think_ this could just be `ProtocolVersion::MAX`. Otherwise LGTM

You were right, this is the right way! Thanks for the suggestion.

Reply:
next time I'll have to put the diff I wasn't sure what you meant yesterday but figured it was caused by me not being clear enough 😅

Reply:
Not at all, I misread it: instead of `ProtocolVersion` I read it `ProtocolConfig` so things did not click :D